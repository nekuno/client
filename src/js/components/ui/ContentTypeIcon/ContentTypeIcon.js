import PropTypes from 'prop-types';
import React, { Component } from 'react';
import styles from './ContentTypeIcon.scss';
import RoundedIcon from "../RoundedIcon/RoundedIcon";

export default class ContentTypeIcon extends Component {

    static propTypes = {
        type : PropTypes.string.isRequired
    };

    constructor(props) {
        super(props);

        this.renderIcon = this.renderIcon.bind(this);
    }

    renderIcon() {
        switch (this.props.type) {
            case 'Audio':
                return <RoundedIcon icon={'music'} size={'small'}/>;
            case 'Video':
                return <RoundedIcon icon={'play'} size={'small'}/>;
            case 'Image':
                return <RoundedIcon icon={'camera'} size={'small'}/>;
            case 'Creator':
                return <RoundedIcon icon={'share-2'} size={'small'}/>;
            case 'Game':
                return <RoundedIcon icon={'steam'} size={'small'}/>;
            default:
                return <RoundedIcon icon={'link'} size={'small'}/>;
        }
    };


    render() {
        return (
            <div className={styles.contentTypeIcon} >
                {this.renderIcon()}
            </div>
        );
    }
}