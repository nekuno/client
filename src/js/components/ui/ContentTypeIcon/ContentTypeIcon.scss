@import '../../../../scss/_partials/colors';

.content-type-icon {
  text-align: center;
  width: 100%;
  z-index: 11111;
  vertical-align: top;
  opacity: 0.7
}