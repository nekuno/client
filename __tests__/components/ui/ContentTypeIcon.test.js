import React from 'react';
import ContentTypeIcon from '../../../src/js/components/ui/ContentTypeIcon/ContentTypeIcon.js';
import styles from '../../../src/js/components/ui/ContentTypeIcon/ContentTypeIcon.scss';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Test ContentTypeIcon component', () => {
    const mockFn = jest.fn();
    const contenttypeicon = shallow(<ContentTypeIcon onClickHandler={mockFn}>Lorem ipsum</ContentTypeIcon>);

    it('should be defined', () => {
        expect(contenttypeicon).toBeDefined();
    });
    it('should render correctly', () => {
        expect(contenttypeicon).toMatchSnapshot();
    });
    it('ContentTypeIcon click event', () => {
        contenttypeicon.find(`.${styles.contenttypeicon}`).simulate('click');
        expect(mockFn).toHaveBeenCalledTimes(1);
    });
});
