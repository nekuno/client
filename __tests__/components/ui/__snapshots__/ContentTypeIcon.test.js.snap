// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ContentTypeIcon component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ContentTypeIcon
    background="#6C6F82"
    onClickHandler={[MockFunction]}
  >
    Lorem ipsum
  </ContentTypeIcon>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <RoundedIcon
        background="#6C6F82"
        icon="link"
        size="small"
      />,
      "className": "contentTypeIcon",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "background": "#6C6F82",
        "icon": "link",
        "size": "small",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <RoundedIcon
          background="#6C6F82"
          icon="link"
          size="small"
        />,
        "className": "contentTypeIcon",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "background": "#6C6F82",
          "icon": "link",
          "size": "small",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
